src/main/java/build/buildfarm/worker/shard/RemoteCasWriter.java

```diff
@@ -114,20 +115,19 @@ public void insertBlob(Digest digest, ByteString content)
   }
 
   private String getRandomWorker() throws IOException {
-    synchronized (workerSet) {
-      if (workerSet.isEmpty()) {
-        throw new IOException("no available workers");
-      }
-      Random rand = new Random();
-      int index = rand.nextInt(workerSet.size());
-      // best case no allocation average n / 2 selection
-      Iterator<String> iter = workerSet.iterator();
-      String worker = null;
-      while (iter.hasNext() && index-- >= 0) {
-        worker = iter.next();
-      }
-      return worker;
+    Set<String> workerSet = backplane.getStorageWorkers();
```

Comment:
will it cause frequent access to backplane like redis ?  How about adding some logic like
```
if (this.workerSet.isEmpty() || System.currentTimeMillis() > this.lastWorkerUpdateTime + 5000) {
    this.lastWorkerUpdateTime = System.currentTimeMillis();
    this.workerSet = backplane.getStorageWorkers();
}

```

Reply:
No, this will not cause frequent call to backplace. 
You may read this code to understand better https://github.com/bazelbuild/bazel-buildfarm/blob/938c789dcdd8603eb6bb1d0525a8d891dbb077c0/src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java#L703

Reply:
I see, cool!